### PR TITLE
[MetaSchedule] Refactor RewriteTensorize to prevent concurrent map updates

### PR DESCRIPTION
**Pull request by vinx13**
Schedule primitives may cause module or prim func to be modified during `sch->Replace`. It is unsafe to iterate the functions in the module and call schedule primitives at the same time. This PR refactored `RewriteTensorize` to decompose it into two parts: find the blocks to rewrite first, and then do the actual rewrite.

cc @masahi @junrushao1994

### CODE REVIEW

**Comment by masahi** [comment]
I thought I've already split visiting and rewriting in `ApplyTensorization`. Is the problem in the iteration over 

```
  for (const auto& kv : sch->mod()->functions) {
       ApplyTensorization(...)
  }
```
?

**Comment by vinx13** [comment]
It seems the map `mod->functions` got `CopyOnWrite`, as a result, the iterator in for loop got invalidated

**Comment by masahi** [comment]
How about returning and collecting `jobs` from `ApplyTensorization`, and do 

```
  for (auto kv : jobs) {
    tir::BlockRV block = sch->GetBlock(kv.first, func_name);
    sch->Unannotate(block, tir::attr::meta_schedule_auto_tensorize);
    kv.second(block);
  }
```

after visiting all `sch->mod()->functions`?

**Review by masahi — APPROVED** [review]
I prefer a solution with less change, but I'm fine with this too.

**Comment by vinx13** [comment]
> How about returning and collecting `jobs` from `ApplyTensorization`, and do
> 
> ```
>   for (auto kv : jobs) {
>     tir::BlockRV block = sch->GetBlock(kv.first, func_name);
>     sch->Unannotate(block, tir::attr::meta_schedule_auto_tensorize);
>     kv.second(block);
>   }
> ```
> 
> after visiting all `sch->mod()->functions`?

that should also work

**Comment by junrushao1994** [comment]
@vinx13 We intentionally create a different IRModule for different threads (see `DeepCopyIRModule`). I was a little bit worried that we missed anything given this approach doesn't seem to work. Would you mind sharing more?

**Comment by vinx13** [comment]
It not related to multi threading. Even if I applied this postproc directly the iterator got invalidated.

File: src/meta_schedule/postproc/rewrite_tensorize.cc [hunk 1]
```diff
@@ -28,26 +28,26 @@ namespace meta_schedule {
 using tir::BlockRV;
 using tir::LoopRV;
 
-void ApplyTensorization(const tir::Schedule& sch, const String& func_name,
-                        const tir::PrimFuncNode* func, bool vectorize_init_loop) {
-  std::vector<std::pair<std::string, std::function<void(tir::BlockRV)>>> jobs;
-
+void ApplyTensorization(
```
Comment (masahi) [inline src/meta_schedule/postproc/rewrite_tensorize.cc:31]: We should rename this function, since we are not applying here.

**Pull request merged**